### DISCUSSION TITLE
Search box query returning empty results

### DISCUSSION THREAD

**Discussion by jnhrv** [microsoft-search/pnp-modern-search#1527] [Q&A]
Hi,
I have configured a Search solution that searches for specific items in the list.

In default view, results are shows correctly:
![image](https://user-images.githubusercontent.com/86712571/142010201-481c959f-9cb7-4c75-bf36-f38d24d6491c.png)

When the search box is used, results return empty:
![image](https://user-images.githubusercontent.com/86712571/142010284-31673555-473b-48d8-9b7d-4ab8041e3dee.png)

Query template:
![image](https://user-images.githubusercontent.com/86712571/142011047-15933de1-433a-47fd-8ecc-f97c85f01543.png)

Search Results connections:
![image](https://user-images.githubusercontent.com/86712571/142011971-0757c5ab-3ffd-4e76-85cb-7d97da371e74.png)

The list and search page are located on a sub-site.
Content-type, columns, crawled properties (site collection level), functional filters/refiners - all works.
Results and Filters are connected.
Search Box using default settings.

Any idea what I missed?
I've configured the modern search on different sites with no issue and see no difference in the approach and config. :(

Thanks.

**Comment by wobba** [comment]
Can you use the debug template and check the value of the queryModification field coming back?

**Reply by jnhrv** [reply]
`"queryModification": "Argentina\n(Path:\"https://XXXXXXXXXX.sharepoint.com/sites/Education/O365/Lists/XXX%20Users%20Country%20DDLs\"\n\ncontentclass:STS_ListItem\n\nContentTypeId:0x00A7470EADF4194E2E9ED1031B61DA0884020050*) -ContentClass=urn:content-class:SPSPeople",`

Not sure where "-ContentClass=urn:content-class:SPSPeople" is coming from and if that is supposed to be there?

**Reply by wobba** [reply]
The last part comes from the default result source using in SharePoint. Nothing to worry about. But seems your query is correct: `Argentina (Path:"https://XXXXXXXXXX.sharepoint.com/sites/Education/O365/Lists/XXX%20Users%20Country%20DDLs"  contentclass:STS_ListItem  ContentTypeId:0x00A7470EADF4194E2E9ED1031B61DA0884020050*) -ContentClass=urn:content-class:SPSPeople`  

And there are no results coming back in the rest of the response? Trying to figure out if you get zero results, or if it's an issue that results are not rendered for some reason. Any errors in the console in the browser?

**Reply by jnhrv** [reply]
No, there are no items in the results.
![image](https://user-images.githubusercontent.com/86712571/142368871-6430bc7d-d810-456d-8c1c-c63b7e801fae.png)

Checked the browser console - only "200" responses, no errors.
![image](https://user-images.githubusercontent.com/86712571/142369323-c58f3e83-b4f7-4c42-9128-b252a51edb90.png)

I remember when I was building the search page, the search actually worked for some time but I don't remember when it stopped working. I guess until properties got crawled and filters were available, it worked. But don't have a clue what has changed.
I've also tried removing the filter webpart - no change. Filtering works.
Next, I tried building the same search on a new page - same result.

Could this be related to managed properties or selected properties in the Search Result?

**Reply by wobba** [reply]
You can try to remove the line breaks in the query template and see if it makes a difference. Or copy paste the query into the SharePoint Search Query Tool and test.

**Reply by jnhrv** [reply]
Just tried the SharePoint Search Query Tool - no luck..
Tested the tool with a query for another modern search page I have - that was showing required results.
![image](https://user-images.githubusercontent.com/86712571/142418397-bbcaf474-f52c-43c4-8f9f-48ad84b2fbf9.png)

**Reply by wobba** [reply]
If it's failing in the query tool, then your query is not returning results. You can try replacing %20 with spaces ` `.

That said, you should add a space before the `(` to make it syntactically correct. But it's likely parsing correct either way.

**Reply by jnhrv** [reply]
Thank you for trying to help me so far! 
Tested with spaces instead of %20 and added a space before the `(`. - No joy.
Anyway, why would the query be incorrect if the query works when is entered in the Query template field of Search Result config?
If I delete the query from the Query template, then I got empty results as well - that should be fine I guess. There is no input in the search box at the same time. When I add the same query (no search phrase of course), then I got the full list of all items which is what I want by default.

Thinking that would be better to start from scratch again.

Sorry for the dumb questions, I'm still learning with this and SP search in general..

EDIT:
If I run this query in SP Seach Query Tool `(Path:"https://XXXXX.sharepoint.com/sites/Education/O365/Lists/XXXx%20Users%20Country%20DDLs" contentclass:STS_ListItem ContentTypeId:0x00A7470EADF4194E2E9ED1031B61DA0884020050*)` I got all results I need.

Changing the query again to `Argentina (Path:"https://XXXXX.sharepoint.com/sites/Education/O365/Lists/XXXx%20Users%20Country%20DDLs" contentclass:STS_ListItem ContentTypeId:0x00A7470EADF4194E2E9ED1031B61DA0884020050*)` no results again...

**Reply by wobba** [reply]
Not sure how to troubleshoot this.. unless the fields/crawled properties containing `Argentina` somehow is mapped to managed properties which are not searchable. Could be you are hitting the note of rule #3 https://www.techmikael.com/2019/10/breaking-sharepoint-search-since-2013.html
